Ferry.java

```diff
@@ -34,7 +34,7 @@ public Island getDestinationIsland() {
     public String processPass(Pass pass, Island island1, Island island2) {
         if (pass.getLuxuryRating() < island2.getIslandRating()) {
             return "The pass has a lower luxury rating than the destination island";
-        } else if (!island1.canPassEnter()) {
+        } else if (!island2.canPassEnter()) {
```

Comment:
Good bug fix

Reply:
yea that's what  I like when you work in pairs, the reviewing part